### PR TITLE
Add Support for site__in and site__not_in parameters

### LINKED ISSUE #1781: Does the ‘sites’ parameter allow negative int for exclusion?
Can I do
```
new WP_Query( array(
    's'     => 'search phrase',
    'sites' => array( -2, -3 ),
) );
```
to exclude site 2 and 3?

### PR DESCRIPTION
**Pull request by burhandodhy**
### Description of the Change

This PR adds the support for the `site__in` and `site__not_in` for the  Post, Term and Comment Query. The old `sites` parameter will throw the deprecated warnings but keep working as it used to. 

Closes #1781

### Changelog Entry

> Added - `site__in` and `site__not_in` parameter support for Post, Term and Comment Query.
> Deprecated - `sites` parameter is set to depreciated for. Post, Term and Comment Query.

### Credits

Props @burhandodhy 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: includes/classes/Indexable/Comment/QueryIntegration.php [hunk 1]
```diff
@@ -116,8 +116,24 @@ public function maybe_filter_query( $results, WP_Comment_Query $query ) {
...
+			$scope    = 'all' === $query->query_vars['sites'] ? 'all' : $site__in;
+		}
+
+		if ( ! empty( $query->query_vars['site__in'] ) ) {
+			$site__in = (array) $query->query_vars['site__in'];
+			$scope    = 'all' === $query->query_vars['site__in'] ? 'all' : $site__in;
+		}
+
+		if ( ! empty( $query->query_vars['site__not_in'] ) ) {
+			$site__not_in = ! is_array( $query->query_vars['site__not_in'] ) ? array( $query->query_vars['site__not_in'] ) : $query->query_vars['site__not_in'];
```
Comment (felipeelia) [inline includes/classes/Indexable/Comment/QueryIntegration.php]: can't we just cast it right away @burhandodhy ? Like 
```
$site__not_in = (array) $query->query_vars['site__not_in'];
```

File: tests/php/indexables/TestPostMultisite.php [hunk 2]
```diff
@@ -1770,4 +1770,242 @@ public function testArchivedSite() {
...
+
+		$this->assertTrue( $query->elasticsearch_success );
+		$this->assertEquals( $query->post_count, 2 );
+		$this->assertEquals( $query->found_posts, 2 );
+
+		$this->cleanUpSites( $sites );
+	}
+
+	/**
+	 * Test a simple post content search on a subset of network sites with depreciated `sites` parameter
```
Comment (felipeelia) [inline tests/php/indexables/TestPostMultisite.php]: There is a typo here @burhandodhy: `depreciated` should be `deprecated`

File: tests/php/indexables/TestPostMultisite.php [hunk 3]
```diff
@@ -1770,4 +1770,242 @@ public function testArchivedSite() {
...
+
+		$this->assertTrue( $query->elasticsearch_success );
+		$this->assertEquals( $query->post_count, 4 );
+		$this->assertEquals( $query->found_posts, 4 );
+
+		$this->cleanUpSites( $sites );
+	}
+
+	/**
+	 * Test a simple post content search with depreciated `sites` parameter
```
Comment (felipeelia) [inline tests/php/indexables/TestPostMultisite.php]: Same typo here

File: tests/php/indexables/TestPostMultisite.php [hunk 4]
```diff
@@ -1770,4 +1770,242 @@ public function testArchivedSite() {
...
+
+			restore_current_blog();
+		}
+
+		$args = array(
+			's'     => 'findme',
+			'sites' => 'all',
+		);
+
+		add_action( 'ep_wp_query_search', array( $this, 'action_wp_query_search' ), 10, 0 );
```
Comment (felipeelia) [inline tests/php/indexables/TestPostMultisite.php]: Why is this needed?

**Review by felipeelia — CHANGES_REQUESTED** [review]
EXCELLENT work on the unit tests, @burhandodhy, thanks. I've left some comments here and there with some minor changes.

One thing we will need to address (and this became obvious with your new code) is that single-site and multisite tests are always run. It would be great if we could split those up and really only run the multisite tests in multisite.

In order to parallelize these tests executions, we could create a new workflow (simply duplicating [this XML](https://github.com/10up/ElasticPress/blob/develop/.github/workflows/test.yml)) and have single tests running in one and multisite tests running in the other. What do you think? Is that something you want to tackle in this PR?

In addition to that change, we will also want to exclude "Multisite.php" classes in single-site.xml.dist

**Comment by burhandodhy** [comment]
> One thing we will need to address (and this became obvious with your new code) is that single-site and multisite tests are always run. It would be great if we could split those up and really only run the multisite tests in multisite.
> 
> In order to parallelize these tests executions, we could create a new workflow (simply duplicating [this XML](https://github.com/10up/ElasticPress/blob/develop/.github/workflows/test.yml)) and have single tests running in one and multisite tests running in the other. What do you think? Is that something you want to tackle in this PR?
> 
> In addition to that change, we will also want to exclude "Multisite.php" classes in single-site.xml.dist

I removed the multisite tests from the `single-site.xml.dist`. We already have another file `phpunit.xml.dist` which runs the all the tests including the multisite tests. Do we still need a separate file for only multisite tests?

File: tests/php/indexables/TestPostMultisite.php (hunk 4 above)
Comment (burhandodhy) [reply tests/php/indexables/TestPostMultisite.php]: Following the pattern of previous tests. We don't need this any more. Will create a new tickets to remove this line from the old tests

File: tests/php/indexables/TestPostMultisite.php (hunk 2 above)
Comment (burhandodhy) [reply tests/php/indexables/TestPostMultisite.php]: Done.

File: includes/classes/Indexable/Comment/QueryIntegration.php (hunk 1 above)
Comment (burhandodhy) [reply includes/classes/Indexable/Comment/QueryIntegration.php]: Done

File: tests/php/indexables/TestPostMultisite.php (hunk 3 above)
Comment (burhandodhy) [reply tests/php/indexables/TestPostMultisite.php]: Fixed

**Review by felipeelia — APPROVED** [review]
